applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClient.java

```diff
@@ -92,7 +95,10 @@ public static ConfigurationAsyncClientBuilder builder() {
 
         // This service method call is similar to setSetting except we're passing If-Not-Match = "*". If the service
         // finds any existing configuration settings, then its e-tag will match and the service will return an error.
-        return service.setKey(serviceEndpoint, setting.key(), setting.label(), setting, null, getETagValue(ETAG_ANY));
+        return service.setKey(serviceEndpoint, setting.key(), setting.label(), setting, null, getETagValue(ETAG_ANY))
+            .doOnRequest(ignoredValue -> logger.asInformational().log("Adding ConfigurationSetting - %s", setting))
+            .doOnSuccess(response -> logger.asInformational().log("Added ConfigurationSetting - %s", response.value()))
+            .doOnError(error -> logger.asWarning().log("Failed to add ConfigurationSetting - %s", setting, error));
```

Comment:
Have you tested this logging works and that the variables are inserted into the output? I note in the [API docs for slf4j](https://www.slf4j.org/api/org/slf4j/Logger.html) that they use `{}` as a placeholder, rather than `%s`

Reply:
It does work but I will switch to using the `{}` since that is what their documentation says.